internal/consensus/common_test.go

```diff
@@ -686,7 +686,32 @@ func ensurePrevote(t *testing.T, voteCh <-chan tmpubsub.Message, height int64, r
 	t.Helper()
 	ensureVote(t, voteCh, height, round, tmproto.PrevoteType)
 }
+func ensurePrevoteMatch(t *testing.T, voteCh <-chan tmpubsub.Message, height int64, round int32, hash []byte) {
+	t.Helper()
+	ensureVoteMatch(t, voteCh, height, round, hash, tmproto.PrevoteType)
+}
+func ensureVoteMatch(t *testing.T, voteCh <-chan tmpubsub.Message, height int64, round int32, hash []byte, voteType tmproto.SignedMsgType) {
+	t.Helper()
+	select {
+	case <-time.After(ensureTimeout):
```

Comment:
should we also pass a context to this? 

(I also get edgy about time.After, because they don't cleanup the goroutine on the other side of the channel until the timeout fires, which might trigger the leak detector)

Reply:
This is the current way the other `ensure*` methods work. Selfishly, I'd prefer to wait until after the PBTS merge happens in the next few days so that I don't have to do another round of conflict fixes and update all these `ensure*` functions in one go.

Reply:
https://github.com/tendermint/tendermint/issues/7649